packages/light-client/src/index.ts

```diff
@@ -392,31 +392,34 @@ export class Lightclient {
    * This headerUpdate may update the head if there's enough participation.
    */
   private processOptimisticUpdate(headerUpdate: altair.LightClientOptimisticUpdate): void {
-    const {attestedHeader, syncAggregate} = headerUpdate;
+    const {attestedHeader, syncAggregate, signatureSlot} = headerUpdate;
 
     // Prevent registering updates for slots to far ahead
     if (attestedHeader.slot > slotWithFutureTolerance(this.config, this.genesisTime, MAX_CLOCK_DISPARITY_SEC)) {
       throw Error(`header.slot ${attestedHeader.slot} is too far in the future, currentSlot: ${this.currentSlot}`);
     }
 
-    const period = computeSyncPeriodAtSlot(attestedHeader.slot);
-    const syncCommittee = this.syncCommitteeByPeriod.get(period);
+    const attestedPeriod = computeSyncPeriodAtSlot(attestedHeader.slot);
+    const signaturePeriod = computeSyncPeriodAtSlot(signatureSlot);
+    // if at sync period boundary, the signature is constructed by sync committee in the next period
+    const syncCommittee = this.getSigningSyncCommitteeAtBoundary(attestedPeriod, signaturePeriod);
```

Comment:
we shouldn't be doing any switching here,
should always use signaturePeriod
```suggestion
    const syncCommittee = this.syncCommitteeByPeriod.get(signaturePeriod);
```